.github/workflows/tests.yml

```diff
@@ -38,7 +38,7 @@ jobs:
     - name: Running serial tests
```

Comment:
can you revert the modifications in this file please?

Reply:
Cool, thanks so much for the review!

I can certainly rearrange the tests, but could you explain a bit about the current structure so that I can implement it correctly? In particular, should the doctests execute before or after the serial/parallel tests?

I don't understand why the tests are currently split into serial/parallel, and from what I can see they are executing in exactly the same context, so I'm confused. By default, unless there's some good reason, I'd expect the tests to run all together in the same job, so that if some of the tests fail then you see all the failing tests rather than only the failing tests in the particular job.

Reply:
The tests are split into serial and parallel because the serial tests can interfere with other tests and make them fail.